sbi/neural_nets/classifier.py

```diff
@@ -0,0 +1,170 @@
+            super().__init__()
+            self.embedding_net_x = embedding_net_x
+            self.embedding_net_y = embedding_net_y
+
+        def forward(self, t):
+            numel = t[0].numel()
+            out = torch.cat(
+                [
+                    self.embedding_net_x(t[:, : numel // 2]),
+                    self.embedding_net_y(t[:, numel // 2 :]),
+                ],
+                dim=1,
+            )
+            return out
+
+    input_layer = StandardizeInputs(embedding_net_x, embedding_net_y)
+
+    # xxx
+
+    return input_layer
+
+
+def build_linear_classifier(
+    batch_x: Tensor = None,
+    batch_y: Tensor = None,
+    z_score_x: bool = True,
+    z_score_y: bool = True,
+) -> nn.Module:
+    """Builds linear classifier for SNRE
+
+    Args:
+        batch_x: Batch of xs, used to infer dimensionality and (optional) z-scoring
+        batch_y: Batch of ys, used to infer dimensionality and (optional) z-scoring
+        z_score_x: Whether to z-score xs passing into the network
+        z_score_y: Whether to z-score ys passing into the network
+
+    Returns:
+        Neural network
+    """
+    x_numel = batch_x[0].numel()
+    y_numel = batch_y[0].numel()
+
+    neural_net = nn.Linear(x_numel + y_numel, 1)
+
+    input_layer = build_input_layer(batch_x, batch_y, z_score_x, z_score_y)
+
+    neural_net = nn.Sequential(input_layer, neural_net)
+
+    return neural_net
+
+
+def build_mlp_classifier(
+    batch_x: Tensor = None,
+    batch_y: Tensor = None,
+    z_score_x: bool = True,
+    z_score_y: bool = True,
+    hidden_features: int = 50,
+) -> nn.Module:
+    """Builds MLP classifier for SNRE
+
+    Args:
+        batch_x: Batch of xs, used to infer dimensionality and (optional) z-scoring
+        batch_y: Batch of ys, used to infer dimensionality and (optional) z-scoring
+        z_score_x: Whether to z-score xs passing into the network
+        z_score_y: Whether to z-score ys passing into the network
+        hidden_features: Number of hidden features
+
+    Returns:
+        Neural network
+    """
+    x_numel = batch_x[0].numel()
+    y_numel = batch_y[0].numel()
+
+    neural_net = nn.Sequential(
+        nn.Linear(x_numel + y_numel, hidden_features),
+        nn.BatchNorm1d(hidden_features),
+        nn.ReLU(),
+        nn.Linear(hidden_features, hidden_features),
+        nn.BatchNorm1d(hidden_features),
+        nn.ReLU(),
+        nn.Linear(hidden_features, 1),
+    )
+
+    input_layer = build_input_layer(batch_x, batch_y, z_score_x, z_score_y)
+
+    neural_net = nn.Sequential(input_layer, neural_net)
+
+    return neural_net
+
+
+def build_resnet_classifier(
+    batch_x: Tensor = None,
+    batch_y: Tensor = None,
+    z_score_x: bool = True,
+    z_score_y: bool = True,
+    hidden_features: int = 50,
+) -> nn.Module:
+    """Builds ResNet classifier for SNRE
+
+    Args:
+        batch_x: Batch of xs, used to infer dimensionality and (optional) z-scoring
```

Comment:
As a microscopic nitpick, we decided on having comments and docstrings use full sentences (capitalized and ended in a full stop).